server/core/src/rodsServer.cpp

```diff
@@ -305,6 +312,110 @@ namespace
+                    log::server::trace("Skipping [{}]. No dot separator found.", p);
+                    continue;
+                }
+
+                const auto epoch_seconds = p.substr(slash_pos, first_dot_pos - slash_pos);
+                const auto remaining_millis = p.substr(first_dot_pos + 1, last_dot_pos - (first_dot_pos + 1));
+                const auto pid = p.substr(last_dot_pos + 1);
+                log::server::trace("epoch seconds = [{}], remaining millis = [{}], agent pid = [{}]",
+                                   epoch_seconds, remaining_millis, pid);
+
+                try {
+                    // Convert the epoch value to ISO8601 format.
+                    log::server::trace("Converting epoch seconds to UTC timestamp ...");
+                    using boost::chrono::system_clock;
+                    using boost::chrono::time_fmt;
+                    const auto tp = system_clock::from_time_t(std::stoll(epoch_seconds));
+                    std::ostringstream utc_ss;
+                    utc_ss << time_fmt(boost::chrono::timezone::utc, "%FT%T") << tp;
+
+                    // Read the contents of the file.
+                    std::ifstream file{p};
+                    const auto stacktrace = boost::stacktrace::stacktrace::from_dump(file);
+                    file.close();
+
+                    // 3. Write the contents of the stacktrace file to syslog.
+                    log::server::trace("Sending stacktrace buffer to syslog ...");
+                    irods::experimental::log::server::error({
+                        {"log_message", boost::stacktrace::to_string(stacktrace)},
+                        {"stacktrace_agent_pid", pid},
+                        {"stacktrace_timestamp_utc", fmt::format("{}.{}", utc_ss.str(), remaining_millis)},
+                        {"stacktrace_timestamp_epoch_seconds", epoch_seconds},
+                        {"stacktrace_timestamp_epoch_milliseconds", remaining_millis}
+                    });
+
+                    // 4. Delete the stacktrace file.
+                    //
+                    // We don't want the stacktrace files to go away without making it into the log.
+                    // We can't rely on the log invocation above because of syslog.
+                    // We don't want these files to accumulate for long running servers.
+                    log::server::trace("Removing stacktrace file from disk ...");
+                    fs::remove(entry);
+                }
+                catch (...) {
+                    // Something happened while logging the stacktrace file.
+                    // Leaving the stacktrace file in-place for processing later.
+                    log::server::trace("Caught exception while processing stacktrace file.");
+                }
+            }
+
+            using namespace std::chrono_literals;
+            std::this_thread::sleep_for(3s); // TODO How long should we wait?
```

Comment:
should this have a configuration options?

Reply:
yes...  now gotta name it too...

and then '3' or whatever could be the backstop default if not/poorly defined in the advanced_settings.

Reply:
do we have any suggestions for the config options?

suggestions:
- `stacktrace_logger_sleep_time_in_seconds`
- `stacktrace_file_detector_sleep_time_in_seconds`
- `stacktrace_file_processor_sleep_time_in_seconds`

Reply:
there aren't any big downsides to using a larger sleep time other than the possibility of more files appearing and the fact that the stacktrace would appear later in the log, in a location that is unrelated to when the stacktrace actually happened.

and yes, the files would survive server restarts.

Reply:
- added config option to server_config.json.template
- updated code to use value from config file
- defaults to 3 if not defined or is less than 1